### DISCUSSION TITLE
Portal should not be wrapped in a div when the target is document.body

### DISCUSSION THREAD

**Discussion by ivanjeremic** [solidjs/solid#1865] [Ideas]
Portal should not be wrapped in a `div` when the target is `document.body`.

Same as with `<head>` when the target is directly the `document.body` the portal markup should not be wrapped in a `div` because we want body > now our markup.

If this is not possible for any technical reason (I don't see any), consider adding an option for turning off the wrapper div.

**Comment by nukemonk** [comment]
It might be not ideal, but as a workaround you could alter the style of this div to get what you need. Just to give you some idea:

```ts
  let ref: HTMLDivElement | undefined;

  createEffect(() => {
    if (ref) {
      ref.style.display = "contents";
      // or add a class
    }
  });

  return (
    <Portal ref={ref}>{/* Some component */}</Portal>
);
```
